File: app/frontend/src/lib/models/link.ts
import * as v from 'valibot';
import type { Platform } from './platform';

export type Link = {
	id: number;
	profile_id: number;
	platform_id: number;
	url: string;
	created_at: string;
	updated_at: string;
	platform: Platform;
};

const updateLinkSchema = v.object({
	id: v.optional(v.string()),
	platform_id: v.pipe(v.string(), v.nonEmpty('Please select a platform.')),
	url: v.pipe(
		v.string(),
		v.nonEmpty('Please enter your url.'),
		v.url('The url is badly formatted.')
	)
});

export const updateLinksSchema = v.object({
	links: v.array(updateLinkSchema)
});

export type UpdateLinks = v.InferOutput<typeof updateLinksSchema>;

File: backend/internal/models/link.go
package models

import "time"

type Link struct {
	ID         int        `json:"id" db:"id"`
	ProfileID  int        `json:"profile_id" db:"profile_id"`
	PlatformID int        `json:"platform_id" db:"platform_id"`
	URL        string     `json:"url" db:"url"`
	Position   int        `json:"position" db:"position"`
	Platform   *Platform  `json:"platform" db:"-"`
	CreatedAt  time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt  *time.Time `json:"updated_at" db:"updated_at"`
}
